### ISSUE TITLE
Packer Crash

### ISSUE THREAD

**Issue by p8952** [mitchellh/packer#1906]
Triggered after hitting Ctrl^C during a docker build.

```
!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Packer crashed! This is always indicative of a bug within Packer.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Packer[1] so that we can fix this.

[1]: https://github.com/mitchellh/packer/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! PACKER CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

$ cat crash.log 
2015/02/07 18:45:10 [INFO] Packer version: 0.7.5  9cd66feeacbd9cb318b72eb5ed59428c5b8c37ac+CHANGES
2015/02/07 18:45:10 Packer Target OS/Arch: linux amd64
2015/02/07 18:45:10 Built with Go Version: go1.3.3
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: amazon-chroot = /usr/local/bin/packer-builder-amazon-chroot
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: amazon-ebs = /usr/local/bin/packer-builder-amazon-ebs
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: amazon-instance = /usr/local/bin/packer-builder-amazon-instance
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: digitalocean = /usr/local/bin/packer-builder-digitalocean
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: docker = /usr/local/bin/packer-builder-docker
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: googlecompute = /usr/local/bin/packer-builder-googlecompute
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: null = /usr/local/bin/packer-builder-null
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: openstack = /usr/local/bin/packer-builder-openstack
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: parallels-iso = /usr/local/bin/packer-builder-parallels-iso
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: parallels-pvm = /usr/local/bin/packer-builder-parallels-pvm
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: qemu = /usr/local/bin/packer-builder-qemu
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: virtualbox-iso = /usr/local/bin/packer-builder-virtualbox-iso
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: virtualbox-ovf = /usr/local/bin/packer-builder-virtualbox-ovf
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: vmware-iso = /usr/local/bin/packer-builder-vmware-iso
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: vmware-vmx = /usr/local/bin/packer-builder-vmware-vmx
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: atlas = /usr/local/bin/packer-post-processor-atlas
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: compress = /usr/local/bin/packer-post-processor-compress
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: docker-import = /usr/local/bin/packer-post-processor-docker-import
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: docker-push = /usr/local/bin/packer-post-processor-docker-push
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: docker-save = /usr/local/bin/packer-post-processor-docker-save
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: docker-tag = /usr/local/bin/packer-post-processor-docker-tag
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: vagrant = /usr/local/bin/packer-post-processor-vagrant
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: vagrant-cloud = /usr/local/bin/packer-post-processor-vagrant-cloud
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: vsphere = /usr/local/bin/packer-post-processor-vsphere
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: ansible-local = /usr/local/bin/packer-provisioner-ansible-local
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: chef-client = /usr/local/bin/packer-provisioner-chef-client
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: chef-solo = /usr/local/bin/packer-provisioner-chef-solo
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: file = /usr/local/bin/packer-provisioner-file
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: puppet-masterless = /usr/local/bin/packer-provisioner-puppet-masterless
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: puppet-server = /usr/local/bin/packer-provisioner-puppet-server
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: salt-masterless = /usr/local/bin/packer-provisioner-salt-masterless
2015/02/07 18:45:10 [DEBUG] Discoverd plugin: shell = /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 Detected home directory from env var: /home/peter
2015/02/07 18:45:10 Detected home directory from env var: /home/peter
2015/02/07 18:45:10 Attempting to open config file: /home/peter/.packerconfig
2015/02/07 18:45:10 File doesn't exist, but doesn't need to. Ignoring.
2015/02/07 18:45:10 Packer config: &{DisableCheckpoint:false DisableCheckpointSignature:false PluginMinPort:10000 PluginMaxPort:25000 Builders:map[parallels-iso:/usr/local/bin/packer-builder-parallels-iso parallels-pvm:/usr/local/bin/packer-builder-parallels-pvm vmware-vmx:/usr/local/bin/packer-builder-vmware-vmx amazon-chroot:/usr/local/bin/packer-builder-amazon-chroot amazon-instance:/usr/local/bin/packer-builder-amazon-instance digitalocean:/usr/local/bin/packer-builder-digitalocean googlecompute:/usr/local/bin/packer-builder-googlecompute openstack:/usr/local/bin/packer-builder-openstack amazon-ebs:/usr/local/bin/packer-builder-amazon-ebs null:/usr/local/bin/packer-builder-null docker:/usr/local/bin/packer-builder-docker qemu:/usr/local/bin/packer-builder-qemu virtualbox-iso:/usr/local/bin/packer-builder-virtualbox-iso virtualbox-ovf:/usr/local/bin/packer-builder-virtualbox-ovf vmware-iso:/usr/local/bin/packer-builder-vmware-iso] PostProcessors:map[docker-save:/usr/local/bin/packer-post-processor-docker-save compress:/usr/local/bin/packer-post-processor-compress docker-import:/usr/local/bin/packer-post-processor-docker-import docker-tag:/usr/local/bin/packer-post-processor-docker-tag vagrant:/usr/local/bin/packer-post-processor-vagrant vagrant-cloud:/usr/local/bin/packer-post-processor-vagrant-cloud vsphere:/usr/local/bin/packer-post-processor-vsphere atlas:/usr/local/bin/packer-post-processor-atlas docker-push:/usr/local/bin/packer-post-processor-docker-push] Provisioners:map[puppet-masterless:/usr/local/bin/packer-provisioner-puppet-masterless puppet-server:/usr/local/bin/packer-provisioner-puppet-server salt-masterless:/usr/local/bin/packer-provisioner-salt-masterless shell:/usr/local/bin/packer-provisioner-shell ansible-local:/usr/local/bin/packer-provisioner-ansible-local chef-client:/usr/local/bin/packer-provisioner-chef-client chef-solo:/usr/local/bin/packer-provisioner-chef-solo file:/usr/local/bin/packer-provisioner-file]}
2015/02/07 18:45:10 Setting cache directory: /home/peter/projects/genstall/packer_cache
2015/02/07 18:45:10 Reading template: packer/gentoo-amd64.json
2015/02/07 18:45:10 Detected home directory from env var: /home/peter
2015/02/07 18:45:10 Creating build: gentoo-amd64-docker
2015/02/07 18:45:10 Loading builder: docker
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-builder-docker
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-builder-docker []string{"/usr/local/bin/packer-builder-docker"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-builder-docker
2015/02/07 18:45:10 packer-builder-docker: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-builder-docker: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-builder-docker: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin126111148
2015/02/07 18:45:10 packer-builder-docker: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-builder-docker: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Loading post-processor: docker-import
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-post-processor-docker-import
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-post-processor-docker-import []string{"/usr/local/bin/packer-post-processor-docker-import"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-post-processor-docker-import
2015/02/07 18:45:10 packer-post-processor-docker-import: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-post-processor-docker-import: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-post-processor-docker-import: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin003560175
2015/02/07 18:45:10 packer-post-processor-docker-import: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-post-processor-docker-import: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Loading post-processor: vagrant
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-post-processor-vagrant
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-post-processor-vagrant []string{"/usr/local/bin/packer-post-processor-vagrant"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-post-processor-vagrant
2015/02/07 18:45:10 packer-post-processor-vagrant: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-post-processor-vagrant: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-post-processor-vagrant: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin781533096
2015/02/07 18:45:10 packer-post-processor-vagrant: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-post-processor-vagrant: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Loading provisioner: shell
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-provisioner-shell []string{"/usr/local/bin/packer-provisioner-shell"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin630108858
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Loading provisioner: shell
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-provisioner-shell []string{"/usr/local/bin/packer-provisioner-shell"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin565638741
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Loading provisioner: shell
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-provisioner-shell []string{"/usr/local/bin/packer-provisioner-shell"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin218024282
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 Loading provisioner: shell
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-provisioner-shell []string{"/usr/local/bin/packer-provisioner-shell"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin550088213
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Loading provisioner: shell
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-provisioner-shell []string{"/usr/local/bin/packer-provisioner-shell"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin091270036
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Loading provisioner: shell
2015/02/07 18:45:10 Creating plugin client for path: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 Starting plugin: /usr/local/bin/packer-provisioner-shell []string{"/usr/local/bin/packer-provisioner-shell"}
2015/02/07 18:45:10 Waiting for RPC address for: /usr/local/bin/packer-provisioner-shell
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin minimum port: 10000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin maximum port: 25000
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Plugin address: unix /tmp/packer-plugin550442904
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Waiting for connection...
2015/02/07 18:45:10 packer-provisioner-shell: 2015/02/07 18:45:10 Serving a plugin connection...
2015/02/07 18:45:10 Skipping build 'gentoo-amd64-vbox' because not specified by -only.
2015/02/07 18:45:10 Skipping build 'gentoo-amd64-aws' because not specified by -only.
2015/02/07 18:45:10 ui: gentoo-amd64-docker output will be in this color.
2015/02/07 18:45:10 ui: 
2015/02/07 18:45:10 Build debug mode: false
2015/02/07 18:45:10 Force build: false
2015/02/07 18:45:10 Preparing build: gentoo-amd64-docker
2015/02/07 18:45:10 Waiting on builds to complete...
2015/02/07 18:45:10 Starting build run: gentoo-amd64-docker
2015/02/07 18:45:10 Running builder: docker
2015/02/07 18:45:10 ui: ==> gentoo-amd64-docker: Creating a temporary directory for sharing data...
2015/02/07 18:45:10 ui: ==> gentoo-amd64-docker: Pulling Docker image: ubuntu
2015/02/07 18:45:10 packer-builder-docker: 2015/02/07 18:45:10 Executing: /usr/bin/docker [pull ubuntu]
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: ubuntu:latest: The image you are pulling has been verified
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: 511136ea3c5a: Already exists
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: 27d47432a69b: Already exists
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: 5f92234dcf1e: Already exists
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: 51a9c7c1f8bb: Already exists
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: 5ba9dab47459: Already exists
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: Status: Image is up to date for ubuntu:latest
2015/02/07 18:45:12 ui: ==> gentoo-amd64-docker: Starting docker container...
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: Run command: docker run -v /tmp/packer-docker314938139:/packer-files -d -i -t ubuntu /bin/bash
2015/02/07 18:45:12 packer-builder-docker: 2015/02/07 18:45:12 Starting container with args: [run -v /tmp/packer-docker314938139:/packer-files -d -i -t ubuntu /bin/bash]
2015/02/07 18:45:12 packer-builder-docker: 2015/02/07 18:45:12 Waiting for container to finish starting
2015/02/07 18:45:12 ui:     gentoo-amd64-docker: Container ID: 39d4cf7fbc2d9fe62f09138d65c6e7268ce29dcdea8f4a605399c75ae2fc86c4
2015/02/07 18:45:12 packer-builder-docker: 2015/02/07 18:45:12 Running the provision hook
2015/02/07 18:45:12 ui: ==> gentoo-amd64-docker: Provisioning with shell script: genstall.d/00-settings.sh
2015/02/07 18:45:12 packer-provisioner-shell: 2015/02/07 18:45:12 Opening genstall.d/00-settings.sh for reading
2015/02/07 18:45:12 [INFO] 936 bytes written for 'uploadData'
2015/02/07 18:45:12 packer-provisioner-shell: 2015/02/07 18:45:12 [INFO] 936 bytes written for 'uploadData'
2015/02/07 18:45:12 packer-builder-docker: 2015/02/07 18:45:12 Executing in container 39d4cf7fbc2d9fe62f09138d65c6e7268ce29dcdea8f4a605399c75ae2fc86c4: "(cp /packer-files/upload394944446 /tmp/script.sh) >/packer-files/cmd302623237 2>&1; echo $? >/packer-files/cmd302623237-exit"
2015/02/07 18:45:14 packer-provisioner-shell: 2015/02/07 18:45:14 Retryable error: Error uploading script: Upload failed with non-zero exit status: 1
2015/02/07 18:45:16 [INFO] 936 bytes written for 'uploadData'
2015/02/07 18:45:16 packer-provisioner-shell: 2015/02/07 18:45:16 [INFO] 936 bytes written for 'uploadData'
2015/02/07 18:45:16 packer-builder-docker: 2015/02/07 18:45:16 Executing in container 39d4cf7fbc2d9fe62f09138d65c6e7268ce29dcdea8f4a605399c75ae2fc86c4: "(cp /packer-files/upload811596704 /tmp/script.sh) >/packer-files/cmd392749695 2>&1; echo $? >/packer-files/cmd392749695-exit"
2015/02/07 18:45:18 packer-provisioner-shell: 2015/02/07 18:45:18 Retryable error: Error uploading script: Upload failed with non-zero exit status: 1
2015/02/07 18:45:19 packer-provisioner-shell: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 Stopping build: gentoo-amd64-docker
2015/02/07 18:45:19 Closing stdin because interrupt received.
2015/02/07 18:45:19 packer-provisioner-shell: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-provisioner-shell: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-provisioner-shell: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-post-processor-docker-import: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-provisioner-shell: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-builder-docker: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-builder-docker: 2015/02/07 18:45:19 Cancelling the step runner...
2015/02/07 18:45:19 packer-provisioner-shell: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-post-processor-vagrant: 2015/02/07 18:45:19 Received interrupt signal (count: 1). Ignoring.
2015/02/07 18:45:19 packer-builder-docker: 2015/02/07 18:45:19 Cancelling provisioning due to interrupt...
2015/02/07 18:45:19 Provisioner.Cancel err: unexpected EOF
2015/02/07 18:45:19 /usr/local/bin/packer-provisioner-shell: plugin process exited
2015/02/07 18:45:19 ui: ==> gentoo-amd64-docker: Killing the container: 39d4cf7fbc2d9fe62f09138d65c6e7268ce29dcdea8f4a605399c75ae2fc86c4
2015/02/07 18:45:20 Build cancelled: gentoo-amd64-docker
2015/02/07 18:45:20 packer-builder-docker: panic: interface conversion: interface is nil, not string
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 48 [running]:
2015/02/07 18:45:20 packer-builder-docker: runtime.panic(0x7b4ae0, 0xc2081242c0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/runtime/panic.c:279 +0xf5
2015/02/07 18:45:20 packer-builder-docker: github.com/mitchellh/packer/builder/docker.(*Builder).Run(0xc208043ce0, 0x7fcc40304410, 0xc2080aa160, 0x7fcc40304458, 0xc208026110, 0x7fcc40304488, 0xc2080da000, 0x0, 0x0, 0x0, ...)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/builder/docker/builder.go:79 +0x822
2015/02/07 18:45:20 packer-builder-docker: github.com/mitchellh/packer/packer/rpc.(*BuilderServer).Run(0xc208043d00, 0x1, 0xc2080260b0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/packer/rpc/builder.go:93 +0x34e
2015/02/07 18:45:20 packer-builder-docker: reflect.Value.call(0x77b9c0, 0x7d19d8, 0x0, 0x130, 0x824910, 0x4, 0xc20809a1e0, 0x3, 0x3, 0x0, ...)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/reflect/value.go:563 +0x1210
2015/02/07 18:45:20 packer-builder-docker: reflect.Value.Call(0x77b9c0, 0x7d19d8, 0x0, 0x130, 0xc20809a1e0, 0x3, 0x3, 0x0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/reflect/value.go:411 +0xd7
2015/02/07 18:45:20 packer-builder-docker: net/rpc.(*service).call(0xc208019400, 0xc208031480, 0xc208026050, 0xc20806a360, 0xc208043fc0, 0x707aa0, 0xc20802609c, 0x0, 0xa6, 0x6ef2c0, ...)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:382 +0x1c7
2015/02/07 18:45:20 packer-builder-docker: created by net/rpc.(*Server).ServeCodec
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:453 +0x427
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 16 [select]:
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/yamux.(*Stream).Read(0xc208044000, 0xc2080c1000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:123 +0x382
2015/02/07 18:45:20 packer-builder-docker: bufio.(*Reader).fill(0xc20809a0c0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/bufio/bufio.go:97 +0x1b3
2015/02/07 18:45:20 packer-builder-docker: bufio.(*Reader).ReadByte(0xc20809a0c0, 0x82, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/bufio/bufio.go:199 +0x7e
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*ioDecReader).readn1(0xc2080900f0, 0x160)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/decode.go:90 +0x44
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*msgpackDecDriver).initReadNext(0xc2080aa060)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/msgpack.go:540 +0x48
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*Decoder).decode(0xc20809a120, 0x6eb840, 0xc2080aae80)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/decode.go:635 +0x58
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*Decoder).Decode(0xc20809a120, 0x6eb840, 0xc2080aae80, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/decode.go:630 +0x7c
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*rpcCodec).read(0xc2080bc080, 0x6eb840, 0xc2080aae80, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/rpc.go:88 +0xd6
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*goRpcCodec).ReadRequestHeader(0xc2080bc080, 0xc2080aae80, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/rpc.go:127 +0x5e
2015/02/07 18:45:20 packer-builder-docker: net/rpc.(*Server).readRequestHeader(0xc208031480, 0x7fcc40304140, 0xc2080bc080, 0x0, 0x0, 0xc2080aae80, 0x8f2d00, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:552 +0x9b
2015/02/07 18:45:20 packer-builder-docker: net/rpc.(*Server).readRequest(0xc208031480, 0x7fcc40304140, 0xc2080bc080, 0xc20806a240, 0xc2080aa080, 0x6e7b20, 0x0, 0x0, 0x0, 0x0, ...)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:519 +0xbe
2015/02/07 18:45:20 packer-builder-docker: net/rpc.(*Server).ServeCodec(0xc208031480, 0x7fcc40304140, 0xc2080bc080)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:438 +0x69
2015/02/07 18:45:20 packer-builder-docker: github.com/mitchellh/packer/packer/rpc.(*Server).Serve(0xc208043ca0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/packer/rpc/server.go:149 +0x23a
2015/02/07 18:45:20 packer-builder-docker: main.main()
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/plugin/builder-docker/main.go:14 +0xc5
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 19 [finalizer wait]:
2015/02/07 18:45:20 packer-builder-docker: runtime.park(0x412ce0, 0xa58e50, 0xa57009)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/runtime/proc.c:1369 +0x89
2015/02/07 18:45:20 packer-builder-docker: runtime.parkunlock(0xa58e50, 0xa57009)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/runtime/proc.c:1385 +0x3b
2015/02/07 18:45:20 packer-builder-docker: runfinq()
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/runtime/mgc0.c:2644 +0xcf
2015/02/07 18:45:20 packer-builder-docker: runtime.goexit()
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/runtime/proc.c:1445
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 20 [syscall]:
2015/02/07 18:45:20 packer-builder-docker: os/signal.loop()
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/os/signal/signal_unix.go:21 +0x1e
2015/02/07 18:45:20 packer-builder-docker: created by os/signal.initÂ·1
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/os/signal/signal_unix.go:27 +0x32
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 22 [chan receive]:
2015/02/07 18:45:20 packer-builder-docker: github.com/mitchellh/packer/packer/plugin.funcÂ·017()
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/packer/plugin/server.go:92 +0x62
2015/02/07 18:45:20 packer-builder-docker: created by github.com/mitchellh/packer/packer/plugin.Server
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/packer/plugin/server.go:96 +0xa8f
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 23 [runnable]:
2015/02/07 18:45:20 packer-builder-docker: net.runtime_pollWait(0x7fcc403039a0, 0x72, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/runtime/netpoll.goc:146 +0x66
2015/02/07 18:45:20 packer-builder-docker: net.(*pollDesc).Wait(0xc20802a1b0, 0x72, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/fd_poll_runtime.go:84 +0x46
2015/02/07 18:45:20 packer-builder-docker: net.(*pollDesc).WaitRead(0xc20802a1b0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/fd_poll_runtime.go:89 +0x42
2015/02/07 18:45:20 packer-builder-docker: net.(*netFD).Read(0xc20802a150, 0xc20809f000, 0x1000, 0x1000, 0x0, 0x7fcc403022b8, 0xb)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/fd_unix.go:242 +0x34c
2015/02/07 18:45:20 packer-builder-docker: net.(*conn).Read(0xc208032030, 0xc20809f000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/net.go:122 +0xe7
2015/02/07 18:45:20 packer-builder-docker: bufio.(*Reader).fill(0xc208004240)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/bufio/bufio.go:97 +0x1b3
2015/02/07 18:45:20 packer-builder-docker: bufio.(*Reader).Read(0xc208004240, 0xc208026004, 0xc, 0xc, 0xc, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/bufio/bufio.go:175 +0x230
2015/02/07 18:45:20 packer-builder-docker: io.ReadAtLeast(0x7fcc40303be8, 0xc208004240, 0xc208026004, 0xc, 0xc, 0xc, 0x0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/io/io.go:289 +0xf7
2015/02/07 18:45:20 packer-builder-docker: io.ReadFull(0x7fcc40303be8, 0xc208004240, 0xc208026004, 0xc, 0xc, 0xc, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/io/io.go:307 +0x71
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/yamux.(*Session).recv(0xc208046500)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:332 +0xc9
2015/02/07 18:45:20 packer-builder-docker: created by github.com/hashicorp/yamux.newSession
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:93 +0x247
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 24 [select]:
2015/02/07 18:45:20 ui error: Build 'gentoo-amd64-docker' errored: unexpected EOF
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/yamux.(*Session).send(0xc208046500)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:290 +0x496
2015/02/07 18:45:20 packer-builder-docker: created by github.com/hashicorp/yamux.newSession
2015/02/07 18:45:20 Builds completed. Waiting on interrupt barrier...
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:94 +0x25f
2015/02/07 18:45:20 ui: Cleanly cancelled builds after being interrupted.
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 25 [select]:
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/yamux.(*Session).keepalive(0xc208046500)
2015/02/07 18:45:20 waiting for all plugin processes to complete...
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:246 +0xc7
2015/02/07 18:45:20 packer-builder-docker: created by github.com/hashicorp/yamux.newSession
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:96 +0x289
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 26 [select]:
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/yamux.(*Session).AcceptStream(0xc208046500, 0xc200000001, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/session.go:154 +0x11f
2015/02/07 18:45:20 packer-builder-docker: github.com/mitchellh/packer/packer/rpc.(*muxBroker).Run(0xc208043c80)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/packer/rpc/mux_broker.go:126 +0x36
2015/02/07 18:45:20 packer-builder-docker: created by github.com/mitchellh/packer/packer/rpc.NewServer
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/mitchellh/packer/packer/rpc/server.go:43 +0x77
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 69 [runnable]:
2015/02/07 18:45:20 packer-builder-docker: sync.(*Mutex).Unlock(0xc2080314a0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/sync/mutex.go:82
2015/02/07 18:45:20 packer-builder-docker: net/rpc.(*Server).freeRequest(0xc208031480, 0xc2080aa080)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:495 +0x64
2015/02/07 18:45:20 packer-builder-docker: net/rpc.(*service).call(0xc208019400, 0xc208031480, 0xc208026050, 0xc20806a240, 0xc2080aa080, 0x6e7b20, 0xc208026ed0, 0x0, 0x160, 0x6e7b20, ...)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:390 +0x35e
2015/02/07 18:45:20 packer-builder-docker: created by net/rpc.(*Server).ServeCodec
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/server.go:453 +0x427
2015/02/07 18:45:20 packer-builder-docker: 
2015/02/07 18:45:20 packer-builder-docker: goroutine 49 [runnable]:
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/yamux.(*Stream).Read(0xc2080443c0, 0xc2080d9000, 0x1000, 0x1000, 0x0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/yamux/stream.go:123 +0x382
2015/02/07 18:45:20 packer-builder-docker: bufio.(*Reader).fill(0xc20809a360)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/bufio/bufio.go:97 +0x1b3
2015/02/07 18:45:20 packer-builder-docker: bufio.(*Reader).ReadByte(0xc20809a360, 0x5b6cb4, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/bufio/bufio.go:199 +0x7e
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*ioDecReader).readn1(0xc208090270, 0xc208001aa0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/decode.go:90 +0x44
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*msgpackDecDriver).initReadNext(0xc2080aa120)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/msgpack.go:540 +0x48
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*Decoder).decode(0xc20809a3c0, 0x6eb8a0, 0xc20808f0e0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/decode.go:635 +0x58
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*Decoder).Decode(0xc20809a3c0, 0x6eb8a0, 0xc20808f0e0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/decode.go:630 +0x7c
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*rpcCodec).read(0xc2080bc180, 0x6eb8a0, 0xc20808f0e0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/rpc.go:88 +0xd6
2015/02/07 18:45:20 packer-builder-docker: github.com/hashicorp/go-msgpack/codec.(*goRpcCodec).ReadResponseHeader(0xc2080bc180, 0xc20808f0e0, 0x0, 0x0)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/go/src/github.com/hashicorp/go-msgpack/codec/rpc.go:123 +0x5e
2015/02/07 18:45:20 packer-builder-docker: net/rpc.(*Client).input(0xc20809a420)
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/client.go:109 +0xda
2015/02/07 18:45:20 packer-builder-docker: created by net/rpc.NewClientWithCodec
2015/02/07 18:45:20 packer-builder-docker: 	/Users/mitchellh/code/3rdparty/go/src/pkg/net/rpc/client.go:201 +0x8b
2015/02/07 18:45:20 /usr/local/bin/packer-builder-docker: plugin process exited
2015/02/07 18:45:20 /usr/local/bin/packer-provisioner-shell: plugin process exited
2015/02/07 18:45:20 /usr/local/bin/packer-provisioner-shell: plugin process exited
2015/02/07 18:45:20 /usr/local/bin/packer-post-processor-vagrant: plugin process exited
2015/02/07 18:45:20 /usr/local/bin/packer-provisioner-shell: plugin process exited
2015/02/07 18:45:20 /usr/local/bin/packer-post-processor-docker-import: plugin process exited
2015/02/07 18:45:20 /usr/local/bin/packer-provisioner-shell: plugin process exited
2015/02/07 18:45:20 /usr/local/bin/packer-provisioner-shell: plugin process exited
```

**Comment by p8952** [comment]
Just FYI, I can replicate this crash if needed.

**Comment by ntfc** [comment]
I'm also having this problem with packer 0.7.5 and docker 1.4.1, using a shell provisioner.

(this seems to be the same as #1894)

**Comment by sethvargo** [comment]
Duplicate of https://github.com/mitchellh/packer/issues/1838 :smile:
